src/dags/crowdmonitor.py

```diff
@@ -86,81 +53,143 @@ def copy_data_from_dbwaarnemingen_to_masterdb(*args, **kwargs):
+            SELECT v.sensor,
+                   'uur' AS periode,
+                   s.location_name AS naam_locatie,
+                   v.datum_uur,
+                   v.aantal_passanten,
+                   s.gebied,
+                   ST_Transform(s.geom, 28992) AS geometrie  -- Ref DB only uses SRID 28992
+                FROM cmsa v
+                         JOIN peoplemeasurement_sensors s ON s.objectnummer = v.sensor;
+            """
         )
         while True:
-            fetch_iterator = cursor.fetchmany(size=import_step)
-            batch_count = copy_data_in_batch(fetch_iterator)
+            rows: List[RowProxy] = cursor.fetchmany(size=IMPORT_STEP)
+            batch_count = insert_many(rows, IMPORT_STEP)
             count += batch_count
-            if batch_count < import_step:
+            if batch_count < IMPORT_STEP:
                 break
-        print(f"Imported: {count}")
-
-
-def copy_data_in_batch(fetch_iterator, periode="uur"):
-    items = []
-    for row in fetch_iterator:
-        items.append(
-            "('{sensor}', "
-            "'{location_name}',"
-            "'{periode}',"
-            "TIMESTAMP '{datum_uur}', "
-            "{aantal_passanten},"
-            "{gebied},"
-            "{geometrie})".format(
-                sensor=row[0],
-                location_name=row[1],
-                periode=periode,
-                datum_uur=row[2].strftime("%Y-%m-%d %H:%M:%S"),
-                aantal_passanten=int(row[3]),
-                gebied=f"'{row[4]}'" if row[4] else "NULL",
-                geometrie=f"'{row[5]}'" if row[5] else "NULL",
-            )
-        )
-    result = len(items)
-    if result:
-        items_sql = ",".join(items)
-        insert_sql = (
-            f"INSERT INTO {table_id}_temp "
-            "(sensor, naam_locatie, periode, datum_uur, aantal_passanten, gebied, geometrie) "
-            f"VALUES {items_sql};"
+        logger.info("Number of records imported: %d", count)
+
+
+def insert_many(rows: List[RowProxy], batch_size: Optional[int]) -> int:
+    """Insert rows in batches."""
+    if batch_size is None:
+        batch_size = len(rows)
+
+    if rows:
+        col_names = rows[0].keys()
+        insert_stmt = sql.SQL("INSERT INTO {table_id} ({col_names}) VALUES %s").format(
+            table_id=sql.Identifier(f"{TABLE_ID}{TMP_TABLE_POSTFIX}"),
+            col_names=sql.SQL(", ").join(
+                map(
+                    sql.Identifier,
+                    col_names,
+                )
+            ),
         )
 
         masterdb_hook = PostgresHook()
         try:
-            masterdb_hook.run(insert_sql)
+            with closing(masterdb_hook.get_conn()) as conn:
+                with closing(conn.cursor()) as cur:
+                    logger.info("Executing SQL statement %s", insert_stmt.as_string(conn))
+                    execute_values(cur, insert_stmt, rows, page_size=batch_size)
+                conn.commit()
```

Comment:
I think this should work too:

```suggestion
            with closing(masterdb_hook.get_conn()) as conn, closing(conn.cursor()) as cur:
                logger.info("Executing SQL statement %s", insert_stmt.as_string(conn))
                execute_values(cur, insert_stmt, rows, page_size=batch_size)
                conn.commit()
```

Reply:
Definitely and is much cleaner. Will fix. Thanks.